### PR TITLE
Add deprecations for component.author and metadata.manufacture

### PR DESCRIPTION
**Pull request by andreas-hilti**

### CODE REVIEW

**Comment by andreas-hilti** [comment]
The Obsolete flag causes the xml serialization to ignore the corresponding element; I don't see a neat workaround.
Compare also:
https://stackoverflow.com/questions/331013/obsolete-attribute-causes-property-to-be-ignored-by-xmlserialization

**Comment by mtsfoni** [comment]
The usual fix to make another property for the xml serialization which is not browsable should still work, shouldn't it?

**Comment by andreas-hilti** [comment]
> The usual fix to make another property for the xml serialization which is not browsable should still work, shouldn't it?

Yes, it would (I guess acceptable, but not super neat).

File: src/CycloneDX.Core/Models/Component.cs [hunk 1]
```diff
@@ -108,10 +108,20 @@ public enum ComponentScope
         public List<OrganizationalContact> Authors { get; set; }
         public bool ShouldSerializeAuthors() { return Authors?.Count > 0; }
 
-        [XmlElement("author")]
+        [Obsolete("This will be removed in a future version. Use @.authors or @.manufacturer instead.")]
```
Comment (thompson-tomo) [inline src/CycloneDX.Core/Models/Component.cs:111]: Nit: Should be manufacture like property name.

Comment (andreas-hilti) [reply src/CycloneDX.Core/Models/Component.cs:111]: @thompson-tomo Sorry, I don't understand your comment. The obsolte comment is a one-to-one copy from the specification:
https://cyclonedx.org/docs/1.6/json/#components_items_author

**Comment by mtsfoni** [comment]
Is there still something to do here? The pr is still marked as draft.

**Comment by andreas-hilti** [comment]
> Is there still something to do here? The pr is still marked as draft.

From my point of view, no. However, this depends also on the comment above.

File: src/CycloneDX.Core/Models/Component.cs (hunk 1 above)
Comment (thompson-tomo) [reply src/CycloneDX.Core/Models/Component.cs:111]: interesting that the obsoletion text it has the property "manufacturer" however when u look at the class it is actually "manufacture"

Comment (andreas-hilti) [reply src/CycloneDX.Core/Models/Component.cs:111]: In the metadata, there is both "manfacture" (deprecated) and "manufacturer".
In component, you only have "manufacturer".

**Pull request merged**